src/test/java/reporter/CustomReporter.java

```diff
@@ -0,0 +1,117 @@
+package reporter;
+
+import TestRailIntegration.TestRailIntegration;
+import com.gurock.testrail.APIClient;
+import lombok.SneakyThrows;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.testng.*;
+import org.testng.xml.XmlSuite;
+import slackIntegration.EyesSlack;
+
+import java.util.*;
+
+public class CustomReporter extends TestRailIntegration implements IReporter {
```

Comment:
Reporter inherits utility class TestRailIntegration to get access to his field and methods. It's possible in Java but violates common sense. Please remove inheritance and use TestRailIntegration explicitly

Reply:
we can not use TestRailIntegration without CustomReporter and method result of xmlSuites parameter.
It does not doesn't make sense collectors of result (CustomReporter ) and publishing (TestRailIntegration)

Reply:
Done